### PR TITLE
Improve big picture flow

### LINKED ISSUE #7: Improve commands where introductions.html is moved into public folder.
The pseudo code is not understandable. Ideally we stick to the convention of making one file with in a predictable directory, so we can add a copy-pastable move command to the guide.

### LINKED ISSUE #8: we state 'rails server' without any indication where to type that into
- maybe we can find a generic way to add to any piece of example code where this should go. make console / text editor (and then also which file) input distinct.

### LINKED ISSUE #29: Skip unnecessary moving of the about.html page
Currently we're  moving this page twice going from Part 1 to Part 2. One should be enough to explain what's going on. 

It felt like a lot of overhead (command line, plus context switching) and always seeing the same result (in the browser). it was a bit hard to explain why this was necessary.

### PR DESCRIPTION
**Pull request by wrtsprt**
* closes (at least) https://github.com/brug-be/railsgirls_guide/issues/36
* closes https://github.com/brug-be/railsgirls_guide/issues/29 - we no need for moving the page anymore, we will just copy / paste the content 
* closes https://github.com/brug-be/railsgirls_guide/issues/7
* closes https://github.com/brug-be/railsgirls_guide/issues/8

This PR improves the big picture flow for part 2. We came across some inconsistencies and structural problems in the old version, which basically needed a rewrite iso of small changes. 

The high level flow of the new version is: 
* show rails
* use the scaffold
* then add the about page from part 1 (iso of starting with it and having to awkwardly copy it twice)
* then showing how erb works 
* then adding another model 

Since it's a bigger change, I think the diff is not going to be of much help. It's probably better to just read through the new version. 

The rest of part two is untouched to keep the scope of this PR as small as possible. There are still many things to improve. I will put into separate tickets.

### CODE REVIEW

**Comment by borisrorsvort** [comment]
Since the title seams very large, can you add your vision regarding this Pr in the description, will be easier to review afterwards ;) I would also reach out to @vanakenm as he will most likely be in the next board and propose a direction for the next upcoming events.

**Comment by borisrorsvort** [comment]
26bgm86xngbjkoqxu
Thanks for taking a stab at this btw :)

**Comment by wrtsprt** [comment]
Who's up for reviewing this? @borisrorsvort, @vanakenm, @otagi? Anyone else?

File: part_2_dynamic.md [hunk 1]
```diff
@@ -4,66 +4,95 @@ title: Part 2
...
+cd railsgirls-blog
 {% endhighlight %}
 
-* Have a look how Rails generated a folder structure and a lot of files for you. There are two ways of doing that, in the terminal or with Finder/Explorer. Ask your coach about the details. Once you have seen them, move your ```about.html``` to the public folder.
+Find the project directory in the Finder/Explorer. Have a look how Rails generated a folder structure and a lot of files for you. Ask your coach about the details.
+
+
+## Starting the Rails server
+
+A Rails app can deliver dynamically generated content with the help of the Ruby programming language. This is why we can't just open a file anymore in the browser. We need a Rails server that will allow for all the magic to happen.
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: This is unclear. They even probably dont know what a server is. I would rephrase saying something like: «In order to view the files, we need a server. It will get what info needs to be fetched from the tremplate, then inject the info and create the final file as html». Probably can be rephrased as well.

File: part_2_dynamic.md [hunk 2]
```diff
@@ -4,66 +4,95 @@ title: Part 2
...
+
+You start your Rails server as follows. Open you terminal (in the `railsgirls-blog` folder) and type the following:
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: To start the Rails server, open your terminal...

File: part_2_dynamic.md [hunk 3]
```diff
@@ -4,66 +4,95 @@ title: Part 2
...
 
 {% highlight bash %}
-mv --your-downloads-folder-here--/about.html public/
-(i.e. mv ~/Downloads/about.html public/)
+rails server
 {% endhighlight %}
 
-## Start rails app
+Once it's started up, you can go visit your app already under
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: Once it is started, ... ~~already~~...

File: part_2_dynamic.md [hunk 4]
```diff
@@ -4,66 +4,95 @@ title: Part 2
...
 
-* `rails server`
-* visit [http://localhost:3000/about.html](http://localhost:3000/about.html)
+[http://localhost:3000/](http://localhost:3000/).
 
-# Create pages controller
+It's not much but the page indicates that everything worked fine until now and the server is running.
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: ~~It's not much but~~ The page indicates that everything worked fine ~~until now~~ and the server is running

File: part_2_dynamic.md [hunk 5]
```diff
@@ -4,66 +4,95 @@ title: Part 2
...
 
-* open another terminal window and go to your app directory (`cd railsgirls-app`) to continue
-* `rails generate controller pages`
-* Open the file `config/routes.rb` in your text editor. The first line will look something like that:  `Rails.application.routes.draw do`
-* add the following as the second line of the file
+## The power of Rails: creating a scaffold
 
-{% highlight ruby %}
-get "/pages/about" => 'pages#about'
+One (not so secret) weapon of Rails is its ability to quickly get you started using scaffolds. Since we're trying to create blog here, let's put everything into place to create, update and show our blog articles. To do that type the following commands into your terminal:
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: ....it’s...
Add «To do that type the following commands into your terminal:» on a new line

File: part_2_dynamic.md [hunk 6]
```diff
@@ -4,66 +4,95 @@ title: Part 2
...
 
-{% highlight ruby %}
-layout false
+[http://localhost:3000/](http://localhost:3000/)
 
-def about
-  render file: 'public/about'
-end
-{% endhighlight %}
+you're not going to notice any difference, since we didn't tell our app about what to do when someone hits the home page just yet. For now you still have to know where to look for the articles, which is under
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: You're...we haven’t told...

File: part_2_dynamic.md [hunk 7]
```diff
@@ -98,28 +127,26 @@ Other examples could be the
...
-    <li>Cooking</li>
+    <li>French</li>
+    <li>English</li>
+    <li>Clingon</li>
     <li>Whatever</li>
 </ul>
 {% endhighlight %}
 
-Adding another hobby here means you have to repeat writing the HTML tags each and every time.
+Adding another language here means you have to repeat writing the HTML tags each and every time.
```
Comment (borisrorsvort) [inline part_2_dynamic.md:145]: locale instead of language to avoid potential confusion with programming language?

File: part_2_dynamic.md [hunk 8]
```diff
@@ -128,67 +155,80 @@ end
 
 {% highlight erb %}
 <ul>
-  <% @hobbies.each do |hobby| %>
+  <% @languages.each do |language| %>
```
Comment (borisrorsvort) [inline part_2_dynamic.md:160]: @locales ?

File: part_2_dynamic.md [hunk 9]
```diff
@@ -128,67 +155,80 @@ end
...
 </ul>
 {% endhighlight %}
 
-Don't forget to do the same for languages.
 
-## If you want to go wild ask your coach about the following code:
+### And now from the database
 
-Option 1: Just bring it back, using a each_with_index
+We're ready for the next level of magic: Let's use the database to store the languages, because editing your code every time you learn a new language is for amateurs.
```
Comment (borisrorsvort) [inline part_2_dynamic.md:171]: I get the funny tone but «is for amateurs» is not very inclusive.

File: part_2_dynamic.md [hunk 10]
```diff
@@ -128,67 +155,80 @@ end
...
+def about
+  @year = Time.now.year
+  @languages = Language.all
+end
 {% endhighlight %}
 
-Option 2: Just bring it back, using `first` and `drop`
+Now go check what the page looks like.
+
+Does it look weird?!
```
Comment (borisrorsvort) [inline part_2_dynamic.md]: Not sure I’d put this in the guide. It’s more the thing the coach would say, not the guide

**Review by borisrorsvort — APPROVED** [review]
Feel free to integrate the remarks, other than that LGTM 👍

File: part_2_dynamic.md (hunk 5 above)
Comment (wrtsprt) [reply part_2_dynamic.md]: @borisrorsvort do you mean the `it's` in `of Rails is its ability`? I think it's fine according to https://data.grammarbook.com/blog/pronouns/1-grammar-error/.

The other one: is changed.

File: part_2_dynamic.md (hunk 7 above)
Comment (wrtsprt) [reply part_2_dynamic.md:145]: We make the joke that one of the languages is Ruby and I actually don't mind that.

File: part_2_dynamic.md (hunk 9 above)
Comment (wrtsprt) [reply part_2_dynamic.md:171]: I'll put this into a separate ticket, since I didn't change it in this PR.  I think it would be interesting to have the discussion if we want to build some cheekiness into the guide or not. (https://github.com/brug-be/railsgirls_guide/issues/43)

File: part_2_dynamic.md (hunk 10 above)
Comment (wrtsprt) [reply part_2_dynamic.md]: If we would go into detail and explain what it does, I think this would be valid step in the guide, but I think it's a detail that doesn't deliver a lot of value in understanding how rails works and has confusion potential. I'll take it out.
